api/apps/geoprocessing/src/modules/surface-cost/adapters/extract-pu-cost.ts

```diff
@@ -0,0 +1,46 @@
+import { isDefined } from '@marxan/utils';
+import { FeatureCollection, GeoJSON, Geometry } from 'geojson';
+import { PlanningUnitCost } from '../ports/planning-unit-cost';
+import { PuExtractorPort } from '../ports/pu-extractor/pu-extractor.port';
+
+type Properties = {
+  cost: number;
+  planningUnitId: string;
+};
+type MaybeProperties = Partial<Properties> | undefined | null;
+
+export class ExtractPuCost implements PuExtractorPort {
```

Comment:
I am not too keen on class names based on verbs - why don't we name this `PuCostExtractor`?

Reply:
I think this would also match some of your previous code

Reply:
https://github.com/Vizzuality/marxan-cloud/pull/219/commits/e9ae13e05dc6fddc7d10fb612c7b378de4b495af